### PR TITLE
(BOLT-672, BOLT-673) Add plan variables and trusted data

### PR DESCRIPTION
**Pull request by MikaelSmith**
Include plan variables and trusted data when compiling a manifest block.

### CODE REVIEW

File: lib/bolt/applicator.rb [hunk 1]
```diff
@@ -24,16 +24,28 @@ def apply(args, apply_body, _scope)
         params = args[1]
       end
 
+      # collect plan vars and merge them over target vars
+      plan_vars = scope.to_hash
+      %w[trusted server_facts facts].each { |k| plan_vars.delete(k) }
```
Comment (lucywyman) [inline lib/bolt/applicator.rb]: Not sure I follow how they get merged into target vars?

Comment (lucywyman) [reply lib/bolt/applicator.rb]: Ugh, because I didn't read like 3 more lines. nvm.

**Comment by lucywyman** [comment]
This looks fine to me (beyond the merge conflict) but I'm not sure how to test it. I'll give writing a plan that uses it a shot, but if you have one on hand you've been using let me know!

**Comment by MikaelSmith** [comment]
Updated.

File: lib/bolt/applicator.rb [hunk 2]
```diff
@@ -22,14 +22,22 @@ def catalog_apply_task
       Task.new('apply_catalog', [impl], 'stdin')
     end
 
-    def compile(target, ast)
+    def compile(target, ast, plan_vars)
+      trusted = {
+        'certname' => target.host,
+        'hostname' => target.host,
+        'domain' => target.host,
```
Comment (nicklewis) [inline lib/bolt/applicator.rb]: Using host as domain seems wrong... Should we try to cut it out of the hostname or maybe omit it altogether?

Comment (MikaelSmith) [reply lib/bolt/applicator.rb]: I think I'll try using https://github.com/puppetlabs/puppet/blob/master/lib/puppet/context/trusted_information.rb to generate this.

**Pull request merged**